tests/testthat/test-pipeline.R

```diff
@@ -123,3 +123,111 @@ rxTest({
     expect_true(is(ps1, "rxSolve"))
   })
 })
+
+test_that("drop linCmt() endpoint (#355)", {
+  ui <- function() {
+    ini({
+      tcl <- 1
+      tvc <- 1
+      addSd <- 1
+    })
+    model({
+      cl <- tcl
+      vc <- tvc
+      linCmt() ~ add(addSd)
+    })
+  }
+  expect_error(newmod <- model(ui, -linCmt()~.), NA)
+  expect_equal(
+    newmod$lstExpr,
+    list(
+      str2lang("cl <- tcl"),
+      str2lang("vc <- tvc")
+    )
+  )
+})
+
+test_that("Compartment should not be added to ini (rxode2#336)", {
+  uifun <- function() {
+    ini({
+      a <- 2
+      propSd <- c(0, 0.3)
+    })
+    model({
+      d/dt(tumor) <- - a*tumor
+      tumor ~ prop(propSd)
+    })
+  }
+
+  rx_orig <- rxode2(uifun)
+  rx_mod <-
+    model(
+      rx_orig,
+      d/dt(transit2) <- (tumor - transit2)/a,
+      append = TRUE
+    )
+  expect_equal(rx_mod$state, c("tumor", "transit2"))
+  expect_equal(rx_mod$ini$est, c(2, 0.3))
+  expect_equal(rx_mod$ini$name, c("a", "propSd"))
+})
+
+# Tests of individual functions ####
+
+test_that(".getModelLineEquivalentLhsExpressionDropDdt", {
```

Comment:
These don't work on `::check()` since you are calling non-exported functions.  I typically use the `rx$getModelLineEquivalentLhsExpressionDropDdt` to not catch the codefactor errors.

Reply:
For my system, when the functions are not exported, `::check()` doesn't work.  

That being said I haven't checked this branch (as mentioned above). But I have had to modify some of your pull requests later since they didn't `::check()` on my system or the CI system, so I assume it is true here. 

I will check soon.

Reply:
Well, soon takes awhile.  It seems to check now and `matchesLangTemplate()` is not exported.

To be consistent perhaps we should use `.matchesLangTemplate()` instead.